### PR TITLE
Create list

### PR DESCRIPTION
**Pull request by ZeenLamDev**

### CODE REVIEW

**Comment by chriswhitehouse** [comment]
[As a owner So i could rent out a space I would like to list a space.](https://trello.com/c/4h4o5vHo/2-as-a-owner-so-i-could-rent-out-a-space-i-would-like-to-list-a-space)

File: lib/space.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def initialize(id:, name:, description:, price:, date_available_from:, date_available_to:)
+    @id = id
+    @name = name
+    @description = description
+    @price = price
+    @date_available_from = date_available_from
+    @date_available_to = date_available_to
+  end
+
+  def self.create(name:, description:, price:, date_available_from:, date_available_to:)
```
Comment (chriswhitehouse) [inline lib/space.rb:14]: I think standard convention is to put the class methods before the instance methods in a class. But no biggie.

Comment (chriswhitehouse) [inline lib/space.rb]: i realise there's a cross dependency here, bit shouldn't the .create space method also add a user_id?

File: spec/database_helpers.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
+require 'pg'
```
Comment (chriswhitehouse) [inline spec/database_helpers.rb]: Shouldn't need to require 'pg' as the pg gem is required in the database connection file

File: lib/space.rb (hunk 1 above)
Comment (chriswhitehouse) [inline lib/space.rb]: No biggie, but think standard convention is put class methods above instance method in the class definition.

File: spec/features/space_create_spec.rb [hunk 3]
```diff
@@ -4,9 +4,9 @@
     visit('/spaces/new')
     fill_in('name', with: 'Beautiful test space')
     fill_in('description', with: 'This is a test description')
-    fill_in('available_from', with: 01/01/2021)
-    fill_in('available_to', with: 01/01/2021)
-    fill_in('Price', with: 50)
+    fill_in('date_available_from', with: "2021-01-01")
```
Comment (chriswhitehouse) [inline spec/features/space_create_spec.rb:7]: Nice, this looks good.

File: spec/features/space_create_spec.rb [hunk 4]
```diff
@@ -4,9 +4,9 @@
...
+    fill_in('date_available_to', with: "2021-01-02")
+    fill_in('price', with: "50")
 
     click_button('confirm')
```
Comment (chriswhitehouse) [inline spec/features/space_create_spec.rb]: Minor detail, but should confirm be capitalised?

File: spec/units/space_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+describe Space do
+
+  describe '.create' do
+    it 'creates a new space' do
+      space = Space.create(
+        name: "Test name",
+        description: "Test description",
+        date_available_from: "2021-01-01",
+        date_available_to: "2021-01-02",
+        price: "50")
```
Comment (chriswhitehouse) [inline spec/units/space_spec.rb]: As mentioned above, need to test for Space.create including a user_id.

File: spec/units/space_spec.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+      Space.create(
+        name: "Another name",
+        description: "Another description",
+        date_available_from: "2021-01-11",
+        date_available_to: "2021-02-12",
+        price: "50")
+
+      expect(space).to be_a Space
+      # expect(space.id).to eq persisted_data.first['id']
```
Comment (chriswhitehouse) [inline spec/units/space_spec.rb]: Don;t think you need the 'first' for this to work.

File: spec/units/space_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      space = Space.create(
+        name: "Test name",
+        description: "Test description",
+        date_available_from: "2021-01-01",
+        date_available_to: "2021-01-02",
+        price: "50")
+
+      persisted_data = persisted_data_retrieve(table: 'spaces', id: space.id)
+
+      Space.create(
```
Comment (chriswhitehouse) [inline spec/units/space_spec.rb:16]: Not sure how this second Space.create block is being used?

File: views/space_add.erb [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
```
Comment (chriswhitehouse) [inline views/space_add.erb]: should this file be called new.erb and sit in a spaces folder? I think that would conform better to the RESTful routes model.

File: app.rb [hunk 9]
```diff
@@ -16,7 +17,7 @@ class MakersBnB < Sinatra::Base
   end
 
   post "/spaces" do
```
Comment (chriswhitehouse) [inline app.rb:19]: should we have a 'get 'spaces/new' route that renders the form? and a post 'spaces' route from the new.erb form?

**Review by chriswhitehouse — COMMENTED** [review]
Cool, thanks guys.

File: spec/units/space_spec.rb (hunk 5 above)
Comment (chriswhitehouse) [reply spec/units/space_spec.rb]: ok

File: spec/database_helpers.rb (hunk 2 above)
Comment (PiperS52) [reply spec/database_helpers.rb]: Removed

File: spec/features/space_create_spec.rb (hunk 4 above)
Comment (PiperS52) [reply spec/features/space_create_spec.rb]: Changed

File: spec/units/space_spec.rb (hunk 7 above)
Comment (PiperS52) [reply spec/units/space_spec.rb:16]: resolved

File: views/space_add.erb (hunk 8 above)
Comment (PiperS52) [reply views/space_add.erb]: Changed

File: lib/space.rb (hunk 1 above)
Comment (chriswhitehouse) [reply lib/space.rb]: will sort later

**Review by chriswhitehouse — COMMENTED** [review]
Looks good

**Comment by chriswhitehouse** [comment]
Thanks both.

**Comment by chriswhitehouse** [comment]
[Code Review](https://trello.com/c/WhEyDIxv/143-code-review)

**Comment by chriswhitehouse** [comment]
[As a owner So i can give details of my space I would like to provide a name, description & price per night](https://trello.com/c/QnduYTAs/5-as-a-owner-so-i-can-give-details-of-my-space-i-would-like-to-provide-a-name-description-price-per-night)

**Comment by chriswhitehouse** [comment]
[As a owner So i can manage availibility I would like to be able to offer a range of availible dates](https://trello.com/c/fjvZLUAw/9-as-a-owner-so-i-can-manage-availibility-i-would-like-to-be-able-to-offer-a-range-of-availible-dates)

**Pull request merged**